.github/workflows/release.yml

```diff
@@ -0,0 +1,119 @@
+name: Release
+
+on:
+  release:
+    types:
+      - prereleased
+      - released
+
+env:
+    ORIGINAL_REPO_NAME: "newrelic/nri-statsd"
+    DOCKER_PLATFORMS: "linux/amd64,linux/arm64"
+    DOCKER_IMAGE_NAME: newrelic/nri-statsd
+
+jobs:
+  integration-tests:
+    name: 🚧 integration tests
+    runs-on: ubuntu-latest
+    strategy:
+      matrix:
+        arch: [  arm64, amd64 ]
+    steps:
+      - uses: actions/checkout@v4
+
+      - name: Set up QEMU
+        uses: docker/setup-qemu-action@v3
+      - name: Set up Docker Buildx
+        uses: docker/setup-buildx-action@v3
+
+      - name: Login to DockerHub
+        uses: docker/login-action@v3
+        with:
+          username: ${{ secrets.OHAI_DOCKER_HUB_ID }}
+          password: ${{ secrets.OHAI_DOCKER_HUB_PASSWORD }}
+
+      - name: Running integration tests
+        run: |
+          make integration-tests-${{ matrix.arch }}
+
+  release-integration:
+    name: Publish container images to registry
+    runs-on: ubuntu-latest
+    needs: [integration-tests]
+    steps:
+        - name: Checkout master
+          uses: actions/checkout@v4
+        
+        - name: Generate version from tag
+          run: |
+            echo "${{ github.event.release.tag_name }}" | grep -E '^v[0-9.]*[0-9]$'
+            IMAGE_VERSION=$(echo "${{ github.event.release.tag_name }}" | sed 's/^v//')
+            echo "IMAGE_VERSION=$IMAGE_VERSION" >> $GITHUB_ENV
+            if ${{ github.event.release.prerelease }}; then
+              echo "DOCKER_IMAGE_TAG=${IMAGE_VERSION}-pre" >> $GITHUB_ENV
+            else
+              echo "DOCKER_IMAGE_TAG=${IMAGE_VERSION}" >> $GITHUB_ENV
+            fi
```

Comment:
I think the `-pre` approach makes sense, it is used in other integrations publishing images.

Even if the release process for nri-prometheus is a bit more complex, it is also adding the `-pre` suffix for pre-release images: https://github.com/newrelic/nri-prometheus/blob/main/build/release.mk#L73-L76

The infrastructure-bundle also adds a suffix for pre-releases: (`rc` in this case): https://github.com/newrelic/infrastructure-bundle/blob/master/.github/workflows/release.yml#L27

We might want to discuss if we should standardize this somehow 🤔 

However, regardless of the name, I think the suffix should be in the `tag` and not in the image name.

Reply:
Modified this t have the suffix in the tag.
Will mention this in the ticket.